### PR TITLE
Fix Issue 15827 -  std.variant.Variant can not be initialized with some struct

### PR DESCRIPTION
**Pull request by MaksimZh**
https://issues.dlang.org/show_bug.cgi?id=15827

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
Needs test case and also Fix issue 15827 in the commit title

**Comment by JackStouffer** [comment]
LGTM

File: std/variant.d [hunk 1]
```diff
@@ -2627,3 +2627,11 @@ unittest
     v = NS2();
     assert(v.get!NS2.foo() == 30);
 }
+
+// Bugzilla 15827
+// Put this outside unittest to avoid "Error: cannot access frame pointer..."
+version(unittest) struct Foo15827 { Variant v; this(Foo15827 v) {} }
```
Comment (DmitryOlshansky) [inline std/variant.d]: Just use static struct Foo15287 and put it inside please

**Comment by MaksimZh** [comment]
OK. Will fix that on Monday.

**Comment by quickfur** [comment]
LGTM, thanks!

**Comment by quickfur** [comment]
Auto-merge toggled on

**Pull request merged**